mozregression/bisector.py

```diff
@@ -287,22 +336,33 @@ def bisect_nightlies(self, good_date, bad_date):
                                   " sure no inbound revision is missed)")
                 infos = {}
                 days = 6
+                too_much_attempt = False
                 first_date = min(handler.good_date, handler.bad_date)
                 while not 'changeset' in infos:
                     days += 1
+                    if days >= 10:
+                        too_much_attempt = True
+                        break
                     prev_date = first_date - datetime.timedelta(days=days)
                     infos = handler.build_data.get_build_infos_for_date(prev_date)
-                if days > 7:
+                if days > 7 and not too_much_attempt:
                     self._logger.info("At least one build folder was"
                                       " invalid, we have to start from"
                                       " %d days ago." % days)
 
                 if not handler.find_fix:
-                    good_rev = infos['changeset']
+                    good_rev = infos.get('changeset')
                     bad_rev = handler.bad_revision
                 else:
                     good_rev = handler.good_revision
-                    bad_rev = infos['changeset']
+                    bad_rev = infos.get('changeset')
+                if bad_rev is None or good_rev is None:
+                    # old nightly builds do not have the changeset information
+                    # so we can't go on inbound. Anyway, these are probably too
+                    # old and won't even exists on inbound.
+                    self._logger.info("Not enough information to go on"
```

Comment:
May as well be more explicit about what's happening: "Not enough changeset information to produce initial inbound regression range. Builds are probably too old."

Also, this should probably be a warning.